plugins/inventory/foreman.py

```diff
@@ -238,10 +293,214 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
+        params["Location".encode('ascii')] = options[self.want_location]
+        params["IPv4".encode('ascii')] = options[self.want_IPv4]
+        params["IPv6".encode('ascii')] = options[self.want_IPv6]
+        params["Facts".encode('ascii')] = options[self.want_facts]
+        params["Host Group".encode('ascii')] = options[self.want_host_group]
+        params["Host Collections".encode('ascii')] = options[self.want_hostcollections]
+        params["Subnet".encode('ascii')] = options[self.want_subnet]
+        params["Subnet v6".encode('ascii')] = options[self.want_subnet_v6]
+        params["Smart Proxies".encode('ascii')] = options[self.want_smart_proxies]
+        params["Content Attributes".encode('ascii')] = options[self.want_content_facet_attributes]
+        params["Host Parameters".encode('ascii')] = options[self.want_host_params]
+        if self.host_filters:
+            params["Hosts".encode('ascii')] = self.host_filters
+        return params
+
+    def _use_inventory_report(self):
+        use_inventory_report = self.get_option('foreman').get('use_reports_api')
+        if not use_inventory_report:
+            return False
+        status_url = "%s/api/v2/status" % self.foreman_url
+        result = self._get_json(status_url)
+        foreman_version = (LooseVersion(result.get('version')) >= LooseVersion('1.24.0'))
+        return foreman_version
+
+    def _post_request(self):
+        url = "%s/ansible/api/v2/ansible_inventories/schedule" % self.foreman_url
+        session = self._get_session()
+        params = {'input_values': self._fetch_params()}
+        self.poll_interval = self.get_option('report').get('poll_interval')
+        ret = session.post(url, json=params)
+        if not ret:
+            raise Exception("Error scheduling inventory report on foreman. Please check foreman logs!")
+        url = "{0}/{1}".format(self.foreman_url, ret.json().get('data_url'))
+        response = session.get(url)
+        while response:
+            if response.status_code != 204:
+                break
+            sleep(self.poll_interval)
+            response = session.get(url)
+        if not response:
+            raise Exception("Error receiving inventory report from foreman. Please check foreman logs!")
+        else:
+            return response.json()
+
     def _populate(self):
+        if self._use_inventory_report():
+            self._populate_report_api()
+        else:
+            self._populate_host_api()
+
+    def _populate_report_api(self):
+        self.groups = dict()
+        self.hosts = dict()
+        try:
+            inventory_report_response = self._post_request()
+        except Exception:
+            self._populate_host_api()
+            return
+        self.group_prefix = self.get_option('group_prefix')
+        h_data = json.loads(inventory_report_response)
+        null = None  # noqa: F841
+        false = False  # noqa: F841
```

Comment:
what are these for?

Reply:
> eval? no.
These were to help with eval since json returned was ruby and interpreted in python. It wasn't working without these. However, I was trying to use eval since I was facing some issues parsing the returned json from report API that I don't seem to be seeing anymore. Removed these and eval.